cmd/cvc-operator/controller/cstorvolumeclaim.go

```diff
@@ -643,7 +676,349 @@ func addPDBLabelOnCVC(
+// 2. Creates new CVR on new pools only if CVR on that pool doesn't exists. If
+//    CVR already created then do nothing.
+func handleVolumeReplicaCreation(cvc *apis.CStorVolumeClaim, cvObj *apis.CStorVolume) error {
+	pvName := cvc.GetAnnotations()[volumeID]
+	desiredPoolNames := cvclaim.GetDesiredReplicaPoolNames(cvc)
+	newPoolNames := util.ListDiff(desiredPoolNames, cvc.Status.PoolInfo)
+	errs := []error{}
+	var errorMsg string
+
+	svcObj, err := svc.NewKubeClient(svc.WithNamespace(openebsNamespace)).
+		Get(cvc.Name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to get service object %s", cvc.Name)
+	}
+
+	for _, poolName := range newPoolNames {
+		cspiObj, err := cspi.NewKubeClient().
+			WithNamespace(openebsNamespace).
+			Get(poolName, metav1.GetOptions{})
+		if err != nil {
+			errorMsg = fmt.Sprintf("failed to get cstorpoolinstance %s error: %v", poolName, err)
+			errs = append(errs, errors.Errorf("%v", errorMsg))
+			klog.Errorf("%s", errorMsg)
+			continue
+		}
+		hash, err := hash.Hash(pvName + "-" + poolName)
+		if err != nil {
+			errorMsg = fmt.Sprintf(
+				"failed to calculate of hase for new volume replica error: %v",
+				err)
+			errs = append(errs, errors.Errorf("%v", errorMsg))
+			klog.Errorf("%s", errorMsg)
+			continue
+		}
+		// TODO: Add a check for ClonedVolumeReplica scaleup case
+		// Create replica with Recreate state
+		rInfo := replicaInfo{
+			replicaID: hash,
+			phase:     apis.CVRStatusRecreate,
+		}
+		_, err = createCVR(svcObj, cvObj, cvc, cspiObj, rInfo)
+		if err != nil {
+			errorMsg = fmt.Sprintf(
+				"failed to create new replica on pool %s error: %v",
+				poolName,
+				err,
+			)
+			errs = append(errs, errors.Errorf("%v", errorMsg))
+			klog.Errorf("%s", errorMsg)
+			continue
+		}
+	}
+	if len(errs) > 0 {
+		return errors.Errorf("%+v", errs)
+	}
+	return nil
+}
+
+// scaleUpVolumeReplicas does the following work
+// 1. Fetch corresponding CStorVolume object of CVC.
+// 2. Verify is there need to update desiredReplicationFactor of CVC(In etcd).
+// 3. Create CVRs if CVR doesn't exist on scaled cStor
+//    pool(handleVolumeReplicaCreation will handle new CVR creations).
+// 4. If scalingUp volume replicas was completed then do following
+//    things(verifyAndUpdateScaleUpInfo will does following things).
+//    4.1.1 Update PDB according to the new pools(only if volume is HAVolume).
+//    4.1.2 Update PDB label on CVC and replica pool information on status.
+// 5. If scalingUp of volume replicas was not completed then return error
+func scaleUpVolumeReplicas(cvc *apis.CStorVolumeClaim) error {
+	drCount := len(cvc.Spec.Policy.ReplicaPoolInfo)
+	cvObj, err := cv.NewKubeclient().
+		WithNamespace(openebsNamespace).
+		Get(cvc.Name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to get cstorvolumes object %s", cvc.Name)
+	}
+	if cvObj.Spec.DesiredReplicationFactor < drCount {
+		cvObj.Spec.DesiredReplicationFactor = drCount
```

Comment:
should we consider picking replicationCount from CVC instead of CV for cstor-volume-mgmt?
This will avoid maintaining this info at multiple places. So, cstor-volume-mgmt will do like:
- pick the cvc.spec.ReplicaCount if len of ReplicaPoolInfo is zero
- else consider replicationFactor as len of ReplicaPoolInfo if it is present
what steps does cstor-vol-mgmt performs on noticing change in DRF?

Reply:
I tried to find from code, but unable to find other than scale down

Reply:
this info on how cstor-vol-mgmt works helps in review

Reply:
I will update PR with relevant pr's

Reply:
https://github.com/openebs/maya/blob/dd6152021192d2bebcd77487da0b7bdef7249dfe/cmd/cstor-volume-mgmt/controller/volume-controller/handler.go#L483

Comments will explain the case

Reply:
https://github.com/openebs/maya/blob/dd6152021192d2bebcd77487da0b7bdef7249dfe/cmd/cstor-volume-mgmt/controller/volume-controller/handler.go#L534

Scaleup function in cstor-volume-mgmt